src/options.cpp

```diff
@@ -7097,8 +7148,28 @@ void options::OnApplyClick(wxCommandEvent& event) {
   }
 
   g_bPlayShipsBells = pPlayShipsBells->GetValue();
+#ifdef HAVE_PORTAUDIO
   if (pSoundDeviceIndex)
-      g_iSoundDeviceIndex = pSoundDeviceIndex->GetSelection();
+  {
+      if (g_iSoundDeviceIndex != atoi(pSoundDeviceIndex->GetString(pSoundDeviceIndex->GetSelection()).BeforeFirst(':')))
+      {
+          // When not the same device select new and reinit
+          g_anchorwatch_sound->Stop();
+          g_anchorwatch_sound->Close();
+          if (g_PluginSound)
+          {
+              g_PluginSound->Stop();
+              g_PluginSound->Close();
+          }
+      }
+      g_iSoundDeviceIndex = atoi(pSoundDeviceIndex->GetString(pSoundDeviceIndex->GetSelection()).BeforeFirst(':'));
+  }
+  // old : g_iSoundDeviceIndex = pSoundDeviceIndex->GetSelection();
+  if (pSoundvolume)
+      g_SoundVolume = pSoundvolume->GetValue();
+  if (pSoundPlayTime)
+      g_SoundPlayTime = pSoundPlayTime->GetValue();
+#endif
```

Comment:
Does this really need to be a #ifdef? If it's possible to use a dynamic_cast check instead it means that the compiler always will compile (and thus check) the code. Just that advantage is worth the effort.

Reply:
Same, Soundvolume and soundplaytime are only in Portaudio.

Reply:
... Yes "Deutsch"  .... that's the reason about my bad english  . I hope I inserted everthing.
I have not tested yet ! Enough for today. from tomorrow evening it is time for sailing on Baltic sea and not working on PC.

Reply:
Compiler error 
home/travis/build/OpenCPN/OpenCPN/src/chcanv.cpp:6216:52: error: ‘SoundFactory’ was not declared in this scope
     if (dynamic_cast<PortAudioSound*>(SoundFactory())) {

my Compiler 
1>
1>chcanv.cpp
1>Erstellen des Projekts opencpn.vcxproj beendet.
========== Erstellen: 1 erfolgreich, 0 fehlerhaft, 0 aktuell, 0 übersprungen ==========

what's wrong ?

Reply:
It's indeed defined in SoundFactory.h :)